### ISSUE TITLE
Allow `__all__` to be ignored in similarities check

### ISSUE THREAD

**Issue by arHSM** [PyCQA/pylint#5385]
### Bug description

I have 2 files, `__init__.py` and `struct.py`, both have an `__all__` variable,
pylint marks these as duplicate but in reality these should be ignored.

`__init__.py` contents :
```py
__all__ = (
    "Session",
    "Ready",
    "Resumed",
    "Reconnect",
    # Trimmed lines due to excessive length
    "VoiceServerUpdate",
    "WebhooksUpdate",
)
```
`structs.py` contents :
```py
__all__ = (
    "Ready",
    "Resumed",
    "Reconnect",
    # Trimmed lines due to excessive length
    "VoiceServerUpdate",
    "WebhooksUpdate",
)
```

### Command used

```shell
pylnt.Run(["./discpyth"])

https://github.com/DiscPyth/DiscPyth/blob/master/dutils.py#L67
```

### Pylint output

```shell
************* Module discpyth.event
discpyth/event.py:1:0: R0801: Similar lines in 2 files
==discpyth.structs:[1:59]
==discpyth:[6:64]
    "Ready",
    "Resumed",
    "Reconnect",
    "ChannelCreate",
    "ChannelDelete",
    "ChannelPinsUpdate",
    "ChannelUpdate",
    "ThreadCreate",
    "ThreadDelete",
    "ThreadUpdate",
    "ThreadListSync",
    "ThreadMemberUpdate",
    "ThreadMembersUpdate",
    "GuildCreate",
    "GuildDelete",
    "GuildUpdate",
    "GuildBanAdd",
    "GuildBanRemove",
    "GuildEmojisUpdate",
    "GuildIntegrationsUpdate",
    "GuildStickersUpdate",
    "GuildMemberAdd",
    "GuildMemberRemove",
    "GuildMemberUpdate",
    "GuildMembersChunk",
    "GuildRoleCreate",
    "GuildRoleDelete",
    "GuildRoleUpdate",
    "GuildScheduledEventCreate",
    "GuildScheduledEventDelete",
    "GuildScheduledEventUpdate",
    "GuildScheduledEventUserAdd",
    "GuildScheduledEventUserRemove",
    "IntegrationCreate",
    "IntegrationDelete",
    "IntegrationUpdate",
    "InviteCreate",
    "InviteDelete",
    "InteractionCreate",
    "MessageCreate",
    "MessageUpdate",
    "MessageDelete",
    "MessageDeleteBulk",
    "MessageReactionAdd",
    "MessageReactionRemove",
    "MessageReactionRemoveAll",
    "MessageReactionRemoveEmoji",
    "PresenceUpdate",
    "StageInstanceCreate",
    "StageInstanceDelete",
    "StageInstanceUpdate",
    "TypingStart",
    "UserUpdate",
    "VoiceStateUpdate",
    "VoiceServerUpdate",
    "WebhooksUpdate",
)
 (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 9.98/10 (previous run: 9.98/10, +0.00)
```

### Expected behavior

Do nothing and ignore the values in `__all__`.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6
Python 3.10.0 (default, Nov  8 2021, 23:29:49) [GCC 8.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
Why is this a false positive according to you ? Isn't it a real duplication with all the issue that come with it ? Wouldn't it be better to choose the ``__init__`` or ``struct`` to be the API of your package ? (Thinking of "There should be one-- and preferably only one --obvious way to do it." here)

**Comment by arHSM** [comment]
Some tools do not "like it" when you dynamically add values to `__all__` i.e
```py
from . import structs

__all__ = ("Session",) + structs.__all__

from .discpyth import Session
from .structs import *
```
Hence, I have some duplicated code.

My use case here isn't enough to justify that pylint should ignore `__all__` during similarity checks, but I surely think that it's something worth being implemented.

**Comment by arHSM** [comment]
Now that I think about it, this seems dumb.
The tool is faulty not pylint, and I cannot disregard the zen for such faulty tools.
